lib/shared/bucketing-assembly-script/assembly/types/target.ts

```diff
@@ -43,65 +43,112 @@ export class Target extends JSON.Value {
     }
 }
 
-export class NoIdAudience extends JSON.Value {
-    filters: TopLevelOperator
+export class AudienceFilter extends JSON.Value {
+    type: string
+    constructor(filter: JSON.Obj) {
+        super()
+        this.type = isValidString(filter, 'type', validTypes, false)
 
-    constructor(audience: JSON.Obj) {
+    }
+
+    stringify(): string {
+        const json = new JSON.Obj()
+        if (this.type) {
+            json.set('type', this.type)
+        }
+
+        return json.stringify()
+    }
+}
+// This is a virtual container class that's used by the code to determine whether a filter is a nested operator
+// or a base level filter. This is used to support the recursive nature of nested filters.
+// The config is returned to its original form when stringified.
+export class AudienceFilterOrOperator extends JSON.Value {
+    operatorClass: AudienceOperator | null
+    filterClass: AudienceFilter | null
+
+    constructor(filter: JSON.Obj) {
+        super()
+        const operator = isValidStringOptional(filter, 'operator', validAudienceOperators)
+        this.operatorClass = operator ? new AudienceOperator(filter) : null
+        this.filterClass = initializeFilterClass(filter)
```

Comment:
this should throw if neither of these get set

Reply:
I guess it always tries to turn it into a filter class, and that will fail

Reply:
it actually wasnt failing in the `initializeFilterClass` function, so I changed that